cartoframes/data/observatory/enrichment/enrichment_service.py

```diff
@@ -83,358 +66,56 @@ def _prepare_data(self, dataframe, geom_col):
-            for f in variable_filter:
-                conditions.append(
-                    _build_where_condition(
-                        variable.column_name,
-                        f
-                    )
-                )
-        else:
-            conditions.append(
-                _build_where_condition(
-                    variable.column_name,
-                    variable_filter
-                )
-            )
-
-        return conditions
-
-
-def _build_where_condition(column, condition):
-    return "enrichment_table.{} {}".format(column, condition)
-
-
-def _build_where_clausule(filters):
-    where = ''
-    if len(filters) > 0:
-        where = 'WHERE {}'.format(' AND '.join(filters))
-
-    return where
-
-
-@timelogger
-def prepare_variables(variables, credentials, aggregation=None):
-    _validate_variables_input(variables)
-
-    if isinstance(variables, list):
-        variables = [_prepare_variable(var, aggregation) for var in variables]
-    else:
-        variables = [_prepare_variable(variables, aggregation)]
-
-    variables = list(filter(None, variables))
-
-    _validate_bq_operations(variables, credentials)
-
-    return variables
-
-
-def _prepare_variable(variable, aggregation=None):
-    if isinstance(variable, str):
-        variable = Variable.get(variable)
-
-    if not isinstance(variable, Variable):
-        raise EnrichmentError("""
-            variable should be a `<cartoframes.data.observatory> Variable` instance,
-            Variable `id` property or Variable `slug` property
-        """)
-
-    if aggregation is not None:
-        variable_agg = _get_aggregation(variable, aggregation)
-        if not variable_agg and aggregation is not AGGREGATION_NONE:
-            log.warning('%s skipped because it does not have aggregation method', variable.id)
-            return None
-
-    return variable
-
-
-def _validate_variables_input(variables):
-    if not isinstance(variables, Variable) and not isinstance(variables, str) and not isinstance(variables, list):
-        raise EnrichmentError('variables parameter should be a Variable instance, a list or a str.')
-
-    if not isinstance(variables, Variable) and len(variables) < 1:
-        raise EnrichmentError('You should add at least one variable to be used in enrichment.')
-
-    if isinstance(variables, list) and len(variables) > MAX_VARIABLES_NUMBER:
-        raise EnrichmentError('The maximum number of variables to be used in enrichment is 50.')
-
-
-def _validate_bq_operations(variables, credentials):
-    dataset_ids = list(set([variable.dataset for variable in variables]))
+    @timelogger
+    def _upload_data(self, temp_table_name, geodataframe):
+        reduced_geodataframe = geodataframe[[_ENRICHMENT_ID, _GEOM_COLUMN]]
 
-    for dataset_id in dataset_ids:
-        dataset = Dataset.get(dataset_id)
-        geography = Geography.get(dataset.geography)
+        dataset = DODataset(auth_client=self.auth_client).name(temp_table_name) \
```

Comment:
This creates an instance of DODataset on every enrichment. Could be better creating an instance in the constructor and then use `self.do_dataset` instead?

Reply:
AFAIK it prevents https://app.clubhouse.io/cartoteam/story/61906/error-500-field-enrichment-id-already-exists-in-schema

Reply:
Yes, "carto dataset" = "bq table", hence it makes all the sense to instantiate a new object cause they don't necessarily have to have the same name, schema, etc.